rpm/0071-sailfishos-webrtc-Implement-video-capture-module.-JB.patch

```diff
@@ -0,0 +1,797 @@
++++ b/media/webrtc/trunk/webrtc/modules/video_capture/BUILD.gn
+@@ -98,10 +98,12 @@ if (!build_with_chromium) {
+ 
+     if (is_linux || is_bsd) {
+       sources = [
+-        "linux/device_info_linux.cc",
+-        "linux/device_info_linux.h",
+-        "linux/video_capture_linux.cc",
+-        "linux/video_capture_linux.h",
++        "gst/device_info_gst.cc",
++        "gst/device_info_gst.h",
++        "gst/video_capture_gst.cc",
++        "gst/video_capture_gst.h",
++        "gst/gst_camera.cpp",
++        "gst/gst_camera.h",
+       ]
+       deps += [ "../..:webrtc_common" ]
+     }
+@@ -211,13 +213,6 @@ if (!build_with_chromium) {
+           "-lm",
+         ]
+       }
+-      if (is_linux) {
+-        ldflags += [
+-          "-lrt",
+-          "-lXext",
+-          "-lX11",
+-        ]
+-      }
+ 
+       deps = [
+         ":video_capture_internal_impl",
+diff --git a/media/webrtc/trunk/webrtc/modules/video_capture/linux/device_info_linux.cc b/media/webrtc/trunk/webrtc/modules/video_capture/linux/device_info_linux.cc
+index cb8bb6fe6e50..d1bcdf2d3c33 100644
+--- a/media/webrtc/trunk/webrtc/modules/video_capture/linux/device_info_linux.cc
++++ b/media/webrtc/trunk/webrtc/modules/video_capture/linux/device_info_linux.cc
+@@ -241,7 +241,6 @@ uint32_t DeviceInfoLinux::NumberOfDevices()
+     WEBRTC_TRACE(webrtc::kTraceApiCall,
+                  webrtc::kTraceVideoCapture, 0, "%s", __FUNCTION__);
+ 
+-#if !defined(MOZ_EMBEDLITE)
+     uint32_t count = 0;
+     char device[20];
+     int fd = -1;
+@@ -258,14 +257,6 @@ uint32_t DeviceInfoLinux::NumberOfDevices()
+     }
+ 
+     return count;
+-#else
+-    /*
+-     * Video capture is not yet implemented on SFOS.
+-     * Do not report about /dev/video* devices as it will break audio calls
+-     * on sites requesting video devices.
+-     */
+-    return 0;
+-#endif
+ }
+ 
+ int32_t DeviceInfoLinux::GetDeviceName(
+diff --git a/media/webrtc/trunk/webrtc/modules/video_capture/sfos/device_info_sfos.cc b/media/webrtc/trunk/webrtc/modules/video_capture/sfos/device_info_sfos.cc
+new file mode 100644
+index 000000000000..f659c344fcf4
+--- /dev/null
++++ b/media/webrtc/trunk/webrtc/modules/video_capture/sfos/device_info_sfos.cc
+@@ -0,0 +1,141 @@
++/* This Source Code Form is subject to the terms of the Mozilla Public
++ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
++ * You can obtain one at http://mozilla.org/MPL/2.0/. */
++
++#include <errno.h>
++#include <fcntl.h>
++#include <stdio.h>
++#include <stdlib.h>
++#include <sys/ioctl.h>
++#include <sys/stat.h>
++#include <unistd.h>
++
++#include "mozilla/Preferences.h"
++#include "webrtc/system_wrappers/include/logging.h"
++
++#include "webrtc/modules/video_capture/sfos/device_info_sfos.h"
++
++#define EXPECTED_CAPTURE_DELAY_PREF "media.getusermedia.camera.expected_capture_delay"
++#define DROID_CAMERA_PRIMARY    "primary"
++#define DROID_CAMERA_SECONDARY  "secondary"
```

Comment:
Are these used anywhere?

Reply:
These are leftovers from previous implementation. Removed, thank you.